src/NuGet.Core/NuGet.Protocol/HttpSource/HttpSourceAuthenticationHandler.cs

```diff
@@ -198,7 +203,13 @@ private async Task<ICredentials> AcquireCredentialsAsync(HttpStatusCode statusCo
                         _packageSource.Source);
                 }
 
-                var promptCredentials = await PromptForCredentialsAsync(
+                ICredentials promptCredentials = default;
+                if (!areLastKnownGoodCredentialsTried)
+                {
+                    _ = _credentialService.TryGetLastKnownGoodCredentialsFromCache(uri: _packageSource.SourceUri, isProxy: false, out promptCredentials);
```

Comment:
The implementation of [CredentialService.TryGetLastKnownGoodCredentialsFromCache](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Credentials/CredentialService.cs#L160-L183) includes a for loop that checks whether the credentials for the given URL were successfully acquired previously. If true, it returns the cached credentials to the caller; otherwise, it returns false.